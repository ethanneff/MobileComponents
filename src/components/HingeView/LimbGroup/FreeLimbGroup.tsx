import React from "react";
import { G } from "react-native-svg";
import Victor from "victor";
import Theme from "../../../commons/Theme";
import Hinge from "../Hinge";
import Limb from "../Limb";

interface FreeLimbGroupProps {
  startPosition: Victor;
  endPosition: Victor;
  limbId?: string;
  hingeId?: string;
}

const styles = {
  freeLimb: {
    strokeColor: Theme.color.neutral300,
    strokeWidth: 27
  },
  freeHinge: {
    fillColor: Theme.color.neutral0,
    strokeColor: Theme.color.neutral300,
    radius: 22,
    strokeWidth: 12
  }
};

/*
Consists of one limb, with one free hinge (no fill) hinge at the end
*/

export default function FreeLimbGroup({
  startPosition,
  endPosition,
  limbId,
  hingeId
}: FreeLimbGroupProps) {
  return (
    <G>
      <Limb
        id={limbId}
        point1={startPosition}
        point2={endPosition}
        strokeColor={styles.freeLimb.strokeColor}
        strokeWidth={styles.freeLimb.strokeWidth}
      />
      <Hinge
        id={hingeId}
        position={endPosition}
        radius={styles.freeHinge.radius}
        strokeColor={styles.freeHinge.strokeColor}
        strokeWidth={styles.freeHinge.strokeWidth}
        fillColor={styles.freeHinge.fillColor}
      />
    </G>
  );
}
