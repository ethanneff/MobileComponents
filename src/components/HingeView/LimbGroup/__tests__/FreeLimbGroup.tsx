import React from "react";
import Victor from "victor";
import { testRender } from "../../../../commons/Utils/TestMocks/test-render";
import FreeLimbGroup from "../FreeLimbGroup";

describe("FreeLimbGroup Component", () => {
  it("renders correctly", () => {
    const { tree } = testRender(
      <FreeLimbGroup
        limbId="freeLimb"
        startPosition={new Victor(10, 20)}
        endPosition={new Victor(20, 20)}
      />
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
