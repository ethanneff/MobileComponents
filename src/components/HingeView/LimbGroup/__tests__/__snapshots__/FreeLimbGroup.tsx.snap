// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FreeLimbGroup Component renders correctly 1`] = `
<RNSVGGroup
  fill={4278190080}
  fillOpacity={1}
  fillRule={1}
  font={Object {}}
  matrix={
    Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ]
  }
  opacity={1}
  originX={0}
  originY={0}
  propList={Array []}
  rotation={0}
  scaleX={1}
  scaleY={1}
  skewX={0}
  skewY={0}
  stroke={null}
  strokeDasharray={null}
  strokeDashoffset={null}
  strokeLinecap={0}
  strokeLinejoin={0}
  strokeMiterlimit={4}
  strokeOpacity={1}
  strokeWidth={1}
  vectorEffect={0}
  x={0}
  y={0}
>
  <RNSVGLine
    fill={4278190080}
    fillOpacity={1}
    fillRule={1}
    matrix={
      Array [
        1,
        0,
        0,
        1,
        0,
        0,
      ]
    }
    name="freeLimb"
    opacity={1}
    originX={0}
    originY={0}
    propList={
      Array [
        "stroke",
        "strokeWidth",
      ]
    }
    rotation={0}
    scaleX={1}
    scaleY={1}
    skewX={0}
    skewY={0}
    stroke={4287603630}
    strokeDasharray={null}
    strokeDashoffset={null}
    strokeLinecap={0}
    strokeLinejoin={0}
    strokeMiterlimit={4}
    strokeOpacity={1}
    strokeWidth={27}
    vectorEffect={0}
    x={0}
    x1={10}
    x2={20}
    y={0}
    y1={20}
    y2={20}
  />
  <RNSVGCircle
    cx={20}
    cy={20}
    fill={4294967295}
    fillOpacity={1}
    fillRule={1}
    matrix={
      Array [
        1,
        0,
        0,
        1,
        0,
        0,
      ]
    }
    opacity={1}
    originX={0}
    originY={0}
    propList={
      Array [
        "fill",
        "stroke",
        "strokeWidth",
      ]
    }
    r={22}
    rotation={0}
    scaleX={1}
    scaleY={1}
    skewX={0}
    skewY={0}
    stroke={4287603630}
    strokeDasharray={null}
    strokeDashoffset={null}
    strokeLinecap={0}
    strokeLinejoin={0}
    strokeMiterlimit={4}
    strokeOpacity={1}
    strokeWidth={12}
    vectorEffect={0}
    x={0}
    y={0}
  />
</RNSVGGroup>
`;
